tartiflette/types/union.py

```diff
@@ -185,3 +190,44 @@ def bake(self, schema: "GraphQLSchema") -> None:
         for field in self._fields.values():
             field.bake(schema, custom_default_resolver)
             await field.on_post_bake()
+
+
+class GraphQLUnionTypeExtension(GraphQLType, GraphQLExtension):
+    def __init__(self, name, directives, types):
+        self.name = name
+        self.directives = directives
+        self.types = types or []
+
+    def bake(self, schema):
+        extended = schema.type_definitions[self.name]
```

Comment:
```suggestion
        extended = schema.find_type(self.name)
```

Reply:
I always forgot about this method ... maybe we should also make `type_definitions` `_type_definitions` so i'm not tempted to use it outside the schema object :).